packages/@lwc/template-compiler/src/codegen/codegen.ts

```diff
@@ -744,9 +748,13 @@ export default class CodeGen {
                     // TODO [#3658]: `disableSyntheticShadowSupport` should also disable this dynamic behavior
                     const isIdOrIdRef =
                         (name === 'id' || isIdReferencingAttribute(name)) &&
-                        (isExpression(value) || isStringLiteral(value));
+                        !isBooleanLiteral(value);
```

Comment:
The only three possibilities here are 1) expression, 2) string literal, 3) boolean literal.